core/src/main/java/org/apache/gravitino/catalog/CatalogManager.java

```diff
@@ -791,12 +791,17 @@ private static boolean catalogInUse(EntityStore store, NameIdentifier ident)
 
   private static boolean getCatalogInUseValue(EntityStore store, NameIdentifier catalogIdent) {
     try {
-      CatalogEntity catalogEntity =
-          store.get(catalogIdent, EntityType.CATALOG, CatalogEntity.class);
+      CatalogWrapper wrapper = catalogCache.getIfPresent(catalogIdent);
+      CatalogEntity catalogEntity;
+      if (wrapper != null) {
+        catalogEntity = wrapper.catalog.entity();
+      } else {
+        catalogEntity = store.get(catalogIdent, EntityType.CATALOG, CatalogEntity.class);
```

Comment:
Shall we put this `catalogEntity` in cache? Besides, can we use loadcatalog directly?

Reply:
Loading `catalogEntity` and then transform it to standard CatalogEntity are NOT `static` method, so I omit it. in fact, in most cases, we can get `catalogEntity` from the cache as all calls are from catalog operations, and the catalog should be in the cache except the first time.